### PR TITLE
Add option to do collisions every # time steps

### PR DESCRIPTION
**Pull request by Yin-YinjianZhao**
This PR makes the collision module execute every # time steps.
A new input parameter `ndt` is needed from the user.
In `doCoulombCollisions` of `Source/Particles/MultiParticleContainer.cpp`, judgement is made to do or not the collision.
Then, when calling `doCoulombCollisionsWithinTile`, the argument `dt` is changed to `dt*ndt`.

### CODE REVIEW

File: Source/Particles/MultiParticleContainer.cpp [hunk 1]
```diff
@@ -662,12 +662,16 @@ MultiParticleContainer::doFieldIonization (int lev,
...
-MultiParticleContainer::doCoulombCollisions ()
+MultiParticleContainer::doCoulombCollisions ( Real cur_time )
 {
     WARPX_PROFILE("MPC::doCoulombCollisions");
 
     for( auto const& collision : allcollisions )
     {
+
+        const Real dt = WarpX::GetInstance().getdt(0);
+        if ( int(std::floor(cur_time/dt)) % collision->m_ndt != 0 ) continue;
```
Comment (ax3l) [inline Source/Particles/MultiParticleContainer.cpp:691]: Would it be more robust to just modulo on the time step integer?

Do you already have an idea how we want to handle this in mesh-refined patches?

Comment (Yin-YinjianZhao) [reply Source/Particles/MultiParticleContainer.cpp:691]: @ax3l It should be easier and more robust to just modulo on the time step integer, but I don't see it is passed into `WarpX::OneStep_nosub (Real cur_time)`, but `cur_time` is already passed. I'm okay with either way.

I don't have an idea how to handle this in mesh-refined patches.
I feel like the collision module is independent to mesh refinement.

Comment (ax3l) [reply Source/Particles/MultiParticleContainer.cpp:691]: Discussed this with @RemiLehe: `dt` on the refined patch will be automatically smaller so the current implementation will likely be solid. We will need to test this later in more detail.

I am not sure about the stability of the floating point division and floor for very large time steps, but we can cross that bridge when we come to it.

**Review by ax3l — APPROVED** [review]
Thank you for this PR! :rocket: :sparkles:

**Comment by ax3l** [comment]
@Yin-YinjianZhao oh no, little merge conflict. Can you please resolve it? :)

**Comment by Yin-YinjianZhao** [comment]
> @Yin-YinjianZhao oh no, little merge conflict. Can you please resolve it? :)

Just fixed, thanks!

**Review by ax3l — APPROVED** [review]

**Comment by ax3l** [comment]
@Yin-YinjianZhao can you please fix the merge conflict? :) Don't hesitate to tell me if I can help, please.

**Review by ax3l — APPROVED** [review]
Thanks again! :sparkles:

**Pull request merged**